### ISSUE TITLE
When the network env changed, received message will throw error.

### ISSUE THREAD

**Issue by su-chang** [wechaty/wechaty-puppet-padplus#214]
> Important：Please file the issue follow the template, or we won't help you to solve the problem.

## 0. Report Issue Guide

1. Please run the following command and check whether the problem has been fixed：

```
npm install wechaty-puppet-padlus@next
```

2. Please search in the issue first, and make sure your problem had not been reported before

## 1. Versions

- What is your wechaty version?
Answer: 0.38.0

- What is your wechaty-puppet-padplus version?
Answer: 0.7.10

- What is your node version? (run `node --version`)
Answer: 10.16.0

- What os are you using
Answer: Mac OS

## 2. Describe the bug

Give a clear and concise description of what the bug is.

If we change wifi network, the process will restart and when message coming, it will throw error that Database is not open in leveldb.

## 3. To Reproduce

This part is very important: if you can not provide any reproduce steps, then the problem will be very hard to be recognized.

1. run the bot
2. change the wifi network
3. send a message to the bot

## 4. Expected behavior

Give a clear and concise description of what you expected to happen.

Could get the contact or room payload in cache successfully.

## 5. Actual behavior

If applicable, add screenshots to help explain your problem. But do not paste log screenshots here.

Can not open the database.

## 6. Full Output Logs

Set env `WECHATY_LOG=silly` in order to set log level to silly, then we can get the full log (If you dosen't set log env, log level is info as default, we cannot get the full log)

**We need full log instead of log screenshot or log fragments!**

<details>
<summary>
Show Logs
</summary>

```shell
$ WECHATY_LOG=silly node yourbot.js

Question: Paste your FULL(DO NOT ONLY PROVIDE FRAGMENTS) log messages
Answer:
21:16:06 ERR Config ###########################
21:16:06 ERR Config unhandledRejection: ReadError: Database is not open [object Promise]
21:16:06 ERR Config ###########################
21:16:06 ERR Config process.on(unhandledRejection) promise.catch(Database is not open)
Config ReadError: Database is not open
    at maybeError (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/node_modules/levelup/lib/levelup.js:354:32)
    at LevelUP.get (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/node_modules/levelup/lib/levelup.js:185:7)
    at FlashStore.<anonymous> (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/node_modules/flash-store/dist/flash-store.js:181:54)
    at step (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/node_modules/flash-store/dist/config.js:11:19)
    at Object.next (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/node_modules/flash-store/bundles/flash-store.umd.js:4:53)
    at /Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/node_modules/flash-store/dist/flash-store.js:59:34
    at new Promise (<anonymous>)
    at __awaiter (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/node_modules/flash-store/dist/flash-store.js:45:1)
    at FlashStore.get (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/node_modules/flash-store/dist/flash-store.js:171:1)
    at CacheManager.<anonymous> (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/src/server-manager/cache-manager.ts:107:40)
    at Generator.next (<anonymous>)
    at /Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/server-manager/cache-manager.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/server-manager/cache-manager.js:4:12)
    at CacheManager.getContact (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/server-manager/cache-manager.js:84:16)
    at PadplusManager.<anonymous> (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/src/padplus-manager/padplus-manager.ts:1140:45)
    at Generator.next (<anonymous>)
    at /Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/padplus-manager/padplus-manager.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/padplus-manager/padplus-manager.js:4:12)
    at PadplusManager.getContact (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/padplus-manager/padplus-manager.js:986:16)
    at PadplusManager.<anonymous> (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/src/padplus-manager/padplus-manager.ts:1164:32)
    at Generator.next (<anonymous>)
    at /Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/padplus-manager/padplus-manager.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/padplus-manager/padplus-manager.js:4:12)
    at PadplusManager.getContactPayload (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/padplus-manager/padplus-manager.js:1010:16)
    at PuppetPadplus.<anonymous> (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/src/puppet-padplus.ts:363:40)
    at Generator.next (<anonymous>)
    at /Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/puppet-padplus.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/puppet-padplus.js:4:12)
    at PuppetPadplus.contactRawPayload (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/puppet-padplus.js:313:16)
    at PuppetPadplus.<anonymous> (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/src/puppet-padplus.ts:198:18)
    at Generator.next (<anonymous>)
    at /Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/puppet-padplus.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/puppet-padplus.js:4:12)
    at PuppetPadplus.onMessage (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/puppet-padplus.js:148:16)
    at PadplusManager.manager.on.msg (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/src/puppet-padplus.ts:113:39)
    at PadplusManager.on.args (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/src/padplus-manager/padplus-manager.ts:197:18)
    at PadplusManager.emit (events.js:198:13)
    at PadplusManager.EventEmitter.emit (domain.js:448:20)
    at PadplusManager.emit (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/src/padplus-manager/padplus-manager.ts:177:18)
    at PadplusManager.<anonymous> (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/src/padplus-manager/padplus-manager.ts:684:18)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/padplus-manager/padplus-manager.js:5:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:29538) PromiseRejectionHandledWarning: Promise rejection was handled asynchronously (rejection id: 17)
```

</details>

## 7. Additional context

[bug]

**Comment by su-chang** [comment]
Related logs:
```js
21:15:43 ERR GRPC_GATEWAY GRPC SERVER ERROR.
      =====================================================================
      try to reconnect grpc server, waiting...
      =====================================================================

21:15:43 SILL GRPC_GATEWAY err code is : 14, ready to reconnect
21:15:43 SILL GRPC_GATEWAY GRPC Request ApiType: INIT catch error.
21:15:43 SILL GRPC_GATEWAY GRPC Request ApiType: HEARTBEAT catch error.
21:15:48 SILL PadplusManager resetThrottleQueue.subscribe() reason: reconnect
21:15:48 SILL PadplusManager start()
21:15:48 SILL GRPC_GATEWAY stop()
21:15:48 SILL GRPC_GATEWAY GRPC Request ApiType: CLOSE
21:15:48 SILL GRPC_GATEWAY GRPC Request ApiType: CLOSE catch error.
21:15:48 ERR GRPC_GATEWAY can not get heartbeat from grpc server
Error: can not get callback result of INIT
    at PadplusUser.<anonymous> (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/src/padplus-manager/api-request/user.ts:23:13)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/suchang/Desktop/PROJECT/wechaty-related-project/wechaty-puppet-padplus/dist/src/padplus-manager/api-request/user.js:5:58)
21:15:48 ERR Wechaty start() exception: can not get callback result of INIT
==========error==========
Error: can not get callback result of INIT
21:15:48 INFO Wechaty <Puppet#0<PuppetPadplus>(tu)> stop() v0.0.0 is stoping ...
21:15:48 SILL StateSwitch <Wechaty> off() is false
21:15:48 SILL StateSwitch <WechatyReady> off() is false
21:15:48 VERB StateSwitch <WechatyReady> off(true) <- (false)
21:15:48 SILL StateSwitch <Wechaty> off() is false
21:15:48 VERB StateSwitch <Wechaty> off(pending) <- (false)
21:15:48 INFO PuppetPadplus stop()
21:15:48 SILL StateSwitch <PuppetPadplus> off() is false
21:15:48 SILL StateSwitch <PuppetPadplus> off() is false
21:15:48 VERB StateSwitch <PuppetPadplus> off(pending) <- (false)
21:15:48 INFO PuppetPadplus logout(logout in wechaty)
21:15:48 VERB Puppet selfId()
21:15:48 SILL Contact ready() @ Puppet#0<PuppetPadplus>(tu) with id="wxid_orp7dihe2pm112"
21:15:48 SILL Contact ready() isReady() true
21:15:48 SILL Puppet constructor() this.on(reset) payload: "{"data":"padplus reset"}"
21:15:48 SILL Puppet constructor() resetThrottleQueue.subscribe() reason: "padplus reset"
21:15:48 VERB Puppet reset(padplus reset)
==========logout success==========
reason : logout in wechaty
21:15:48 INFO PuppetPadplus stop()
21:15:48 SILL StateSwitch <PuppetPadplus> off() is pending
21:15:48 SILL PuppetPadplus stop() is called on a OFF puppet. await ready(off) and return.
21:15:48 VERB StateSwitch <PuppetPadplus> ready(off, false)
21:15:48 SILL PadplusManager stop()
21:15:48 SILL GRPC_GATEWAY stop()
21:15:48 SILL GRPC_GATEWAY GRPC Request ApiType: CLOSE
21:15:48 SILL GRPC_GATEWAY GRPC Request ApiType: CLOSE catch error.
21:15:53 SILL GRPC_GATEWAY initGrpcGateway()
21:15:53 VERB MemoryCard <puppet> get(WECHATY_PUPPET_PADPLUS)
21:15:53 VERB GRPC_GATEWAY releaseQueue() subscriptions have been released.
21:15:53 VERB GRPC_GATEWAY releaseQueue() queues have been released.
21:15:53 VERB CacheManager release()
21:15:53 VERB CacheManager releaseCache()
21:15:53 SILL CacheManager releaseCache() closing caches ...
21:15:53 SILL PadplusUser initInstance()
21:15:53 SILL RequestClient request()
21:15:53 SILL DedupeApi dedupe() no need to dedupe api 3.
21:15:53 SILL GRPC_GATEWAY GRPC Request ApiType: INIT
21:15:53 SILL ==P==A==D==P==L==U==S==<GRPC DATA>==P==A==D==P==L==U==S==
21:15:53 SILL GRPC_GATEWAY responseType: AUTO_LOGIN, data : {"uin":289099750,"online":true,"wechatUser":{"headImgUrl":"http://wx.qlogo.cn/mmhead/ver_1/QyMgwtU0iccfDf1e159IXBeKW8bw5kiaSiaXicyynj824sBwfPNlOkTB1YFko3p9cTngZrGv5dnz8A4fzThE50wX3JINRO92VLXrTTrbSLFlz3ficL7X5dGiaib4pCyvkrOAvsc/0","nickName":"jessicaTu","uin":289099750,"userName":"wxid_orp7dihe2pm112","alias":"juzi_bot","verifyFlag":0}}
21:15:53 SILL ==P==A==D==P==L==U==S==<GRPC DATA>==P==A==D==P==L==U==S==
```

**Comment by su-chang** [comment]
This bug will be resolved in `v0.7.11`
